tests/interfaces/records/handlers/records-read.spec.ts

```diff
@@ -135,6 +138,189 @@ describe('RecordsReadHandler.handle()', () => {
       expect(Comparer.byteArraysEqual(dataFetched, dataBytes!)).to.be.true;
     });
 
+    describe('protocol based reads', () => {
+      it('should allow read with allow-anyone rule', async () => {
+        // scenario: Alice writes an image to Caroline's DWN with Alice herself as recipient,
+        //           then Bob reads the image because he is "anyone".
+
+        const alice = await DidKeyResolver.generate();
+        const bob = await DidKeyResolver.generate();
+        const caroline = await DidKeyResolver.generate();
+
+        // setting up a stub DID resolver
+        TestStubGenerator.stubDidResolver(didResolver, [alice, bob, caroline]);
```

Comment:
FYI: DID Key doesn't require stub resolver because it is self resolving.

Reply:
Removed